hazelcast/src/test/java/com/hazelcast/config/XMLConfigPreProcessorTest.java

```diff
@@ -500,6 +501,56 @@ public void testXmlDeniesDuplicateMemberAttributesConfig() throws Exception {
         configBuilder.build();
     }
 
+    @Test
+    public void testXmlVariableReplacementAsSubstring() throws Exception {
+        Properties properties = new Properties();
+        String xml = "<hazelcast>\n" +
+                "    <properties>\n" +
+                "        <property name=\"${env}-with-suffix\">local-with-suffix</property>\n" +
+                "        <property name=\"with-prefix-${env}\">with-prefix-local</property>\n" +
+                "    </properties>\n" +
+                "</hazelcast>";
+
+        properties.setProperty("env", "local");
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(bis);
+        configBuilder.setProperties(properties);
+        Config config = configBuilder.build();
+
+        assertEquals(config.getProperty("local-with-suffix"), "local-with-suffix");
+        assertEquals(config.getProperty("with-prefix-local"), "with-prefix-local");
+    }
+
+    @Test
+    public void testXmlImportWithVariableReplacementAsSubstring() throws Exception {
+        Properties properties = new Properties();
```

Comment:
You could even consider something like this (building on top of a varargs of string):

```
    Config config = build(xml, properties, "env", "local");
```

Much cleaner than that you can't get it. Saves you another few non interesting lines from your test.

```
  public void testXmlVariableReplacementAsSubstring() throws Exception {
        String xml = "<hazelcast>\n" +
                "    <properties>\n" +
                "        <property name=\"${env}-with-suffix\">local-with-suffix</property>\n" +
                "        <property name=\"with-prefix-${env}\">with-prefix-local</property>\n" +
                "    </properties>\n" +
                "</hazelcast>";

      Config config = build(xml, "env", "local");

        assertEquals(config.getProperty("local-with-suffix"), "local-with-suffix");
        assertEquals(config.getProperty("with-prefix-local"), "with-prefix-local");
    }
```